### PR TITLE
zipl: make IPL sections defined with BLS to inherit a target field

### PR DESCRIPTION
**Pull request by martinezjavier**
The target field is required for IPL sections, but the ones populated from
BLS snippets did not have this field. This was only working when using BLS
snippets because IPL sections inherit the target field with "defaultauto".

But that's not the case when using a menu, for example the following zipl
configuration will fail if is used along with BLS defined IPL sections:
fail when running the zipl tool:

  [defaultboot]
  defaultmenu = menu1

  :menu1
  target = /boot
  1 = linux
  2 = test
  default = 1
  prompt = 1
  timeout = 0

Because "linux" and "test" will be defined using BLS snippets and these
don't have an option to define target fields. Let's make these to always
inherit a target field, either from the [defaultboot] section or a menu
section if there's a defaultmenu defined.

Fixes: https://github.com/ibm-s390-linux/s390-tools/issues/111

Reported-by: Renaud Métrich <rmetrich@redhat.com>
Signed-off-by: Javier Martinez Canillas <javierm@redhat.com>

### CODE REVIEW

File: zipl/src/scan.c [hunk 1]
```diff
@@ -1358,6 +1262,165 @@ scan_get_section_keywords(struct scan_token* scan, int* index, char* name,
 }
 
 
+/* Get the default target defined either in the defaultboot section or the
+ * menu section for the defaultmenu if there is a default menu defined. */
```
Comment (stefan-haberland) [inline zipl/src/scan.c:1266]: please put the trailing */ on a separate line for block comments

**Review by stefan-haberland — CHANGES_REQUESTED** [review]
Thanks again for this patch.
I tested the code with some scenarios I can imagine and it worked quite well.

The only thing that does not work (but I think it is intended this way) is when you do not have a defaultmenu= set in the zipl.conf but specify the menu on the commandline using "zipl -m menu1" for example. I am also not sure if this can be achieved that easy.

Beside of this I only have some coding style nitpicks. I know that you just copied a lot of the code code and it looks this way all over but maybe we can fix it for the parts you just touched.
General remark: all those "foo* bar" declarations should rather be "foo *bar". Rest of the comments in the patch. Please let me know if you would like to fix the style or if we should the code as is.

best regards,
Stefan

**Comment by hoeppnerj** [comment]
@martinezjavier ping, what is the status of this PR? Are you going to address the last few change requests?

**Comment by hoeppnerj** [comment]
fyi, as there was no response since last year we've pulled the patches and adapted the requested style changes ourselves. Thanks for the contribution!

**Pull request closed without merging**